lib/omniauth/strategies/openid_connect.rb

```diff
@@ -47,8 +47,14 @@ class OpenIDConnect
       option :send_scope_to_token_endpoint, true
       option :client_auth_method
       option :post_logout_redirect_uri
+      option :uid_field, 'sub'
 
-      uid { user_info.sub }
+      def uid
+        user_info.public_send(options.uid_field.to_s)
+      rescue NoMethodError
```

Comment:
I'd be appreciate if you can compare them

Reply:
I have done the unit test calls in for loop with 10000000 iterations and measured the performance with Benchmark:
```
                             user     system      total        real
Missing with if check:  36.439877   0.000153  36.440030 ( 36.502268)
Existing with if check: 36.045783   0.000132  36.045915 ( 36.112107)
Missing with rescue:    36.255928   0.000157  36.256085 ( 36.316388)
Existing with rescue:   36.261213   0.000077  36.261290 ( 36.527901)
```

The results show that there is not a real big difference between the implementations. I let the test run without the log line (to avoid IO dependency) on my laptop which had as much of the processes stopped on it, but still with Xserver

Seeing the results, I would recommend to use the most readable and clean code which in my opinion is the one using `rescue` as it doesn't need a to break the control flow and an extra `if` condition.

Reply:
Interesting results.

Could you please check one more option if you have a chance:

```ruby
user_info.all_attributes.include?(strategy.options.uid_field.to_s.to_sym)
```

Reply:
Here is the data:

```
                               user     system      total        real
Missing with if include:  36.228053   0.001304  36.229357 ( 36.305666)
Existing with if include: 36.146231   0.001040  36.147271 ( 36.209575)
```

Reply:
Thanks a lot for testing this for me, I immensely appreciate this! It looks like you're right and there's no real difference in performance, so it's up to you to decide how to implement fallback.

Reply:
Just let me know when you're ready, I will merge PR immediately

Reply:
I changed back to rescue as it is the cleanest code without and control flow break (next call) and it doesn't require the extra check in case of correct configuration.